sql/src/main/java/com/rilldata/calcite/CalciteToolbox.java

```diff
@@ -185,4 +274,55 @@ public String validateModelingQuery(SqlCreateMetric sqlCreateMetric) throws SqlP
     // dimensions, measures and table are validated, return sql string to be stored in db
     return sqlCreateMetric.toSqlString(sqlDialect).toString();
   }
+
+  private static void createGraph(Map<String, Statement> existing)
+  {
+    for (Statement create : existing.values()) {
+      List<String> dependencies = create.node.accept(new DependencyFinder());
+      create.dependencies = dependencies.stream().map(existing::get).collect(Collectors.toList());
+    }
+  }
+
+  private static Map<String, Statement> toStatementsMap(SqlNode node, SqlDialect sqlDialect)
+  {
+    SqlNodeList list = (SqlNodeList) node;
+    return list.stream().map(n -> {
```

Comment:
- this does not handle Metrics View model ?
- What happens when only name is changed is changed for a ddl statement, looks like it will considered a new statement and computed afresh.

Reply:
If name changes then the current logic should drop previous and create a new entity.